chatmeleon-server/controllers/conversation-controller.ts

```diff
@@ -0,0 +1,158 @@
+        id: conversationId
+      }
+    });
+
+    // Send the conversation data as a JSON response.
+    res.send(JSON.stringify(conversationById));
+  } catch (error) {
+    next(error);
+  } finally {
+    // Disconnect from the Prisma database to release resources.
+    await prisma.$disconnect();
+  }
+};
+
+/* Request body example
+{
+    "lastMessageAt": "2023-10-11T03:23:06.107+00:00",
+    "name": "Test",
+    "isGroup": false,
+    "messagesIds": ["6526151a22f25fa43da0f0f3"],
+    "userIds": ["6524d06b9726a23da4002695"]
+}
+*/
+
+const postConversation = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const {
+      lastMessageAt,
+      name,
+      isGroup,
+      messagesIds,
+      userIds,
+    } = req.body;
+
+    const createdAt = new Date().toISOString();
+
+    const messages = await prisma.message.findMany({
+      where: {
+        id: { in: messagesIds },
+      },
+    })
+
+    const users = await prisma.user.findMany({
+      where: {
+        id: { in: userIds },
+      },
+    })
+
+    const newConversation = await prisma.conversation.create({
+      data: {
+        createdAt,
+        lastMessageAt,
+        name,
+        isGroup,
+        messagesIds,
+        messages: {
+          connect: messages.map((message) => ({ id: message.id })),
+        },
+        userIds,
+        users: {
+          connect: users.map((user) => ({ id: user.id })),
+        },
+      }
+    })
+
+    res.json(newConversation);
+  } catch (error) {
+    next(error);
+  } finally {
+    // Disconnect from the Prisma database to release resources.
+    await prisma.$disconnect();
+  }
+};
+
+const updateConversation = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const conversationId = req.params.id;
+    const updatedConversationData = req.body;
+
+    const updatedConversation = await prisma.conversation.update({
+      where: {
+        id: conversationId,
+      },
+      data: updatedConversationData,
+    });
+
+    res.json(updatedConversation);
+  }
+  catch (error) {
+    next(error);
+  }
+  finally {
+    // Disconnect from the Prisma database to release resources.
+    await prisma.$disconnect();
+  }
+}
+
+const updateConversationFields = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const conversationId = req.params.id;
+    const updatedFields = req.body;
+
+    const updatedConversation = await prisma.conversation.update({
+      where: {
+        id: conversationId,
+      },
+      data: updatedFields,
+    });
+
+    res.json(updatedConversation);
+  } catch (error) {
+    next(error);
+  }
+}
+
+const deleteConversation = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const conversationId = req.params.id;
+
+    await prisma.conversation.delete({
+      where: {
+        id: conversationId,
+      }
+    })
+
+    res.json({ message: `${conversationId} deleted.` });
+  } catch (error) {
+
+  }
+}
```

Comment:
Remove conversation delete route for now. We'll update the API with a new design later.